### DISCUSSION TITLE
How to get the raw ANSI output in terminal

### DISCUSSION THREAD

**Discussion by peterroe** [vuejs/vitepress#3597] [Q&A] [answered]
I noticed an interesting thing about the documentation for this project：

In https://vitepress.dev/guide/getting-started#setup-wizard , Which show the usage of `vitepress`
```
┌  Welcome to VitePress!
│
◇  Where should VitePress initialize the config?
│  ./docs
│
◇  Site title:
│  My Awesome Project
│
◇  Site description:
│  A VitePress Site
│
◆  Theme:
│  ● Default Theme (Out of the box, good-looking docs)
│  ○ Default Theme + Customization
│  ○ Custom Theme
└
```
I found that it is rendered by the following code：

https://github.com/vuejs/vitepress/blob/28989df83446923a9e7c8ada345b0778119ed66f/docs/snippets/init.ansi#L1-L16

I want to achieve the same effect, but display some other content from my terminal

So How can I get the raw ANSI output in terminal like above? 🙏 If anyone know it and share the idea with me, I will be very grateful to you :)

**Comment by brc-dd** [comment]
Depends on what tool is generating that output. But mostly it's `FORCE_COLOR=true your_command 1>foo.ansi`

**Reply by peterroe** [reply]
I got it, thanks! 👍

**Comment by Zhengqbbb** [comment, marked as answer]
**Adding a few points**

---

### 1 `FORCE_COLOR`

**COMMAND**

* Does it support removing colors in non-interactive mode?
* Does it support outputting color characters in `FORCE_COLOR` forced in non-interactive mode?
* Does it output to `stderr`?

**Example:**

```
FORCE_COLOR=true COMMAND >foo.ansi 2>&1
```

https://github.com/vuejs/vitepress/assets/40693636/b15e9260-b936-41a8-8082-bdf92dff65f2




---

### 2 Using the `script` command to record commands

* Use `control+d` to exit recording
* Please note that the output files of the `script` command are different for Linux and **MacOS**.
* Capture in the product file

https://github.com/vuejs/vitepress/assets/40693636/86babf1a-8e25-4e09-8fcb-3d072ce0d206

**Reply by peterroe** [reply]
Thank you very much. It's very helpful to generate the code. And your video is awesome !
